flow/record/adapter/sqlite.py

```diff
@@ -0,0 +1,246 @@
+from __future__ import annotations
+
+import logging
+import sqlite3
+from datetime import datetime
+from typing import Iterator
+
+from flow.record import Record, RecordDescriptor
+from flow.record.adapter import AbstractReader, AbstractWriter
+from flow.record.base import RESERVED_FIELDS
+
+logger = logging.getLogger(__name__)
+
+__usage__ = """
+SQLite adapter
+---
+Write usage: rdump -w sqlite://[PATH]
+Read usage: rdump sqlite://[PATH]
+[PATH]: path to sqlite database file.
+"""
+
+# flow.record field mappings to SQLite types
+FIELD_MAP = {
+    "int": "INTEGER",
+    "uint32": "INTEGER",
+    "varint": "INTEGER",
+    "float": "REAL",
+    "boolean": "INTEGER",
+    "bytes": "BLOB",
+    "filesize": "INTEGER",
+    "datetime": "TIMESTAMP",
+}
+
+
+# SQLite types to flow.record field mappings
+SQLITE_FIELD_MAP = {
+    "VARCHAR": "string",
+    "INTEGER": "varint",
+    "BLOB": "bytes",
+    "REAL": "float",
+    "DOUBLE": "float",
+    "BOOLEAN": "boolean",
+    "DATETIME": "datetime",
+    "TIMESTAMP": "datetime",
+}
+
+
+def sanitized_name(name: str) -> str:
+    """Returns a sanitized version of name.
+
+    Some (field) names are not allowed in flow.record, while they can be allowed in SQLite.
+    This sanitizes the name so it can still be used in flow.record.
+
+    >>> sanitized_name("my-variable-name-with-dashes")
+    'my_variable_name_with_dashes'
+    >>> sanitized_name("_my_name_starting_with_underscore")
+    'n__my_name_starting_with_underscore'
+    >>> sanitized_name("1337")
+    'n_1337'
+    """
+
+    if name not in RESERVED_FIELDS:
+        name = name.replace("-", "_")
+        if name.startswith("_") or name[0].isdecimal():
+            name = "n_" + name
+    return name
+
+
+def create_descriptor_table(con: sqlite3.Connection, descriptor: RecordDescriptor) -> None:
```

Comment:
Would it be an idea to have a "_descriptors" table for databases created with rdump? So you can fully recover the record descriptor? Databases not having it would fall back to using the "type detection".

Reply:
Yes, this was on my mind to implement as some adapters have proper support for this as well. However, I would like to do this in a future PR.